lib/BackgroundJobs/StorageCrawlJob.php

```diff
@@ -78,6 +93,30 @@ protected function run($argument): void {
 			$this->logger->error('Could not find storage root');
 			return;
 		}
+		
+		$qb = new CacheQueryBuilder($this->db, $this->systemConfig, $this->logger);
+		try {
+			$ignoreFiles = $qb->selectFileCache()
+				->andWhere($qb->expr()->eq('mimetype', $qb->createNamedParameter(15)))
+				->andWhere($qb->expr()->eq('name', $qb->createNamedParameter('.nomedia')))
+				->orWhere($qb->expr()->eq('name', $qb->createNamedParameter('.noimage')))
+				->executeQuery()->fetchAll();
+
+			$ignoreDir = [];
+			foreach ($ignoreFiles as $ignoreFile) {
+				$qb = new CacheQueryBuilder($this->db, $this->systemConfig, $this->logger);
+				$parent = $qb->selectFileCache()
+					->andWhere($qb->expr()->eq('mimetype', $qb->createNamedParameter(2)))
+					->andWhere($qb->expr()->eq('filecache.fileid', $qb->createNamedParameter($ignoreFile['parent'])))
+					->executeQuery()->fetch();
+				$ignoreDir[] = $parent;
+				$ignoreDir = array_merge($ignoreDir, $this->getDir($parent['fileid'], true));
+			}
+		} catch (Exception $e) {
```

Comment:
Can we move this block into a method as well, so that we simply have one call to get all ignored directories in the run method?

Reply:
I have made a slight change to the method to get all ignored directories, should I move it to method?

Reply:
Yes, please move to a new method. That way it's cleaner :)